herd/tests/instructions/ASL-pseudo-arch/non-deterministic-optimised.litmus

```diff
@@ -0,0 +1,28 @@
+ASL non-deterministic
+
+{
+  x = 0;
+  0: X1= x;
+}
+
+// Always return 2, but this could create 2 executions that will turn out to
+// follow the same branch. To know if this creates two different executions
+// with the same end state, run herd with -debug solver
```

Comment:
Hi @HadrienRenaud. In case two executions are created, output would be different (with `Positive: 2`). Or do I miss something...

Reply:
OK I have checked. The difference show up with `herd7 -v` . With the optimisation we have a message on stderr: `Warning: 1 abstract event structures`, while without it, we have: `Warning: 2 abstract event structures`.

Reply:
As a test it looks easier to interpret than using `-debug solver`.